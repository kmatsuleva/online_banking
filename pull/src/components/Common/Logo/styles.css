.logo {
    vertical-align: baseline
}

.image {
    width: 250px
}