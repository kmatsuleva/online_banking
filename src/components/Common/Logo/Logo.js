import React from 'react';
import PropTypes from 'prop-types';

const Logo = ({ src, to, alt, className }) => (
	<div className="logo">
		<a href={to}>
			<img src={src} alt={alt} className={className} />
		</a>
	</div>
);

Logo.propTypes = {
	src: PropTypes.string,
	to: PropTypes.string,
	alt: PropTypes.string
};

export default Logo;
