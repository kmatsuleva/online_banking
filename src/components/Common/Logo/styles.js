const styles = {
    logo: {
        vertivalalign: "baseline"
    },

    image: {
        width: 250
    }
};

export default styles

