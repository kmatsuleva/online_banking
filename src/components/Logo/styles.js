const styles = {
    logo: {
        vertivalalign: "baseline",
        display: "inline"
    },

    image: {
        width: 300
    }
};

export default styles

