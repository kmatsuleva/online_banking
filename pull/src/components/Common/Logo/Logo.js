import React from "react";
import PropTypes from 'prop-types';     
import "./styles.css";

const Logo = ({ src, to, alt }) => (
    <div className = "logo">
        <a href = {to} > 
            <img src={src}
                alt = {alt}
                className="image"/>
        </a>
    </div>
);


Logo.propTypes = {
    src: PropTypes.string,
    to: PropTypes.string,
    alt: PropTypes.string
}

export default Logo;
