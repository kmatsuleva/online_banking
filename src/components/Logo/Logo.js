import React from "react";
import styles from "./styles.js";

const Logo = ({ src }) => (
    <div style={styles.logo}>
        <img src={ src }
            style={styles.image}/>
    </div>
);

export default Logo;
