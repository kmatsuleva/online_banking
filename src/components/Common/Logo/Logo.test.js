
import React from 'react'
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Logo from "./Logo"

Enzyme.configure({ adapter: new Adapter() })

describe("< Logo />", () => {

    it('renders without crashing', () => {
        shallow(<Logo />);
    });

    it('should render correctly', () => {
        const output = shallow(
        <Logo  src="../../image.png" to='/' />
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});