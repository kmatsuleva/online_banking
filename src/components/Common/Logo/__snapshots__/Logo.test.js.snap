// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< Logo /> should render correctly 1`] = `
<div
  className="logo"
>
  <a
    href="/"
  >
    <img
      src="../../image.png"
    />
  </a>
</div>
`;
